### ISSUE TITLE
Bug: Error when the content of the link is an image marker

### ISSUE THREAD

**Issue by xyy94813** [quantizor/markdown-to-jsx#529]
Test case:

```mdx
<Markdown>
[![latest version](https://img.shields.io/npm/v/markdown-to-jsx.svg?label=latest%20%20version)](https://www.npmjs.org/package/markdown-to-jsx)
</Markdown>
```

Expected:

[![latest version](https://img.shields.io/npm/v/markdown-to-jsx.svg?label=latest%20%20version)](https://www.npmjs.org/package/markdown-to-jsx)

Actual:

[!\[latest version](https://www.npmjs.org/package/markdown-to-jsx)][(https://www.npmjs.org/package/markdown-to-jsx)](https://www.npmjs.org/package/markdown-to-jsx)

relates to #524

**Comment by quantizor** [comment]
I will be looking into this, thanks!

**Comment by quantizor** [comment]
Fixed via #551, thanks @zegl!
